# HG changeset patch
# Parent  c970bca8297d3a6e348c539b8cb9aa68e19087be
Bug 1184550 - Request constructor should always throw if used flag is set, even if body is null
Moves check for bodyUsed before the check for a null body so subsequent fetches with the same Request fail.
r=bkelly

diff --git a/dom/fetch/Request.cpp b/dom/fetch/Request.cpp
--- a/dom/fetch/Request.cpp
+++ b/dom/fetch/Request.cpp
@@ -233,7 +233,7 @@ Request::Constructor(const GlobalObject&
     inputReq->GetBody(getter_AddRefs(body));
+    if (inputReq->BodyUsed()) {
+      aRv.ThrowTypeError<MSG_FETCH_BODY_CONSUMED_ERROR>();
+      return nullptr;
+    }
     if (body) {
-      if (inputReq->BodyUsed()) {
-        aRv.ThrowTypeError<MSG_FETCH_BODY_CONSUMED_ERROR>();
-        return nullptr;
-      }
       temporaryBody = body;
